### DISCUSSION TITLE
[API Gateway] CRUD Base paths

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#102] [Q&A] [answered]
Hi!

I'm using the console and I have the API Gateway with a single-view exposed under the base path `/my-base-path`.

Now, I would like to expose a MongoDB CRUD collection. By default, the API Gateway appends at the beginning of the base path the prefix `/my-base-path`.

However, If I try to assign the basePath `/my-base-path` to a CRUD collection, that will be exposed as `/v2/my-base-path`, the console would prevent me to assign this basePath with the following message:

```
basePath must be unique
```

Is this a bug or the normal behaviour expected?

In the latter case, is not very clear to me why a base path with `/v2` prefix is considered the same as a base path without prefix.

**Comment by frengooo** [comment]
Hi @giovf96, if I remember the `/v2` prefix is related to crud APIs compatibility issues (v1 APIs had another format, if I'm not wrong).
Probably the console checks for the uniqueness of the `base path` added by the user, without considering the `/v2` prefix.

I think it's a bug, but I'm not sure, because there is the possibility that the `v1-adapter` exposes a `v1` version of crud APIs, to be used with the old CMS. @davidebianchi @fredmaggiowski can tell you more about this behavior 😄

**Comment by davidebianchi** [comment, marked as answer]
Hi @giovf96! This is the normal behaviour, because the endpoint will be exposed on backoffice (so, at the port 8081 of the api-gateway) without the `/v2` prefix.
We are thinking about remove the `/v2` prefix forced by Console, since it has no real value since the path rewrite was introduced. This could be done in a totally retrocompatible way in the next releases, to avoid this unexpected cases.

At the moment, I suggest you to change one of the two exposed endpoints.
